src/renderer/components/cluster-settings/components/cluster-metrics-setting.tsx

```diff
@@ -89,9 +65,26 @@ export class ClusterMetricsSetting extends React.Component<ClusterMetricsSetting
           onMenuClose={this.save}
           closeMenuOnSelect={false}
           controlShouldRenderValue={false}
-          options={Object.values(ClusterMetricsResourceType)}
-          onChange={this.onChangeSelect}
-          formatOptionLabel={this.formatOptionLabel}
+          options={metricResourceTypeOptions}
+          onChange={(options) => {
+            runInAction(() => {
+              this.hiddenMetrics.replace(options.map(opt => opt.type));
+              this.save();
+            });
```

Comment:
Big nit: Business-logic like this does not belong in UI-code, which should be as dumb and pure as possible (Separation of Concerns). Seeing how much code like this we have, there's a lot of work to be done. Let's not add to that? :)